scrapy/utils/misc.py

```diff
@@ -116,4 +116,28 @@ def md5sum(file):
 def rel_has_nofollow(rel):
     """Return True if link rel attribute has nofollow type"""
     return True if rel is not None and 'nofollow' in rel.split() else False
-    
+
+def construct_from_settings_or_crawler(objcls, settings, crawler, *args,
```

Comment:
hm, `from_crawler` is a good factory method name only in a class, it's a bit weird for an util function IMO.